app/liquid_tags/stackexchange_tag.rb

```diff
@@ -0,0 +1,86 @@
+class StackexchangeTag < LiquidTagBase
+  PARTIAL = "liquids/stackexchange".freeze
+  # Filter codes come from the example tools in the docs. For example: https://api.stackexchange.com/docs/posts-by-ids
+  API_URL = "https://api.stackexchange.com/2.2/".freeze
+  FILTERS = {
+    "post" => "!3tz1WbZW5JxrG-f99",
+    "answer" => "!Fcb(61J.xH9ZW2D7KF1bbM_J7X",
+    "question" => "!*1SgQGDOL9bPBHULz9sKS.y6qv7V9fYNszvdhDuv5",
+    "site" => "!mWxO_PNa4i"
+  }.freeze
+
+  attr_reader :site, :post_type
+
+  def initialize(tag_name, input, tokens)
+    super
+    @site = parse_site(input)
+    @post_type = "question"
+    @json_content = get_data(input)
+  end
+
+  def render(_context)
+    ActionController::Base.new.render_to_string(
+      partial: PARTIAL,
+      locals: {
+        site: @site,
+        pretty_site_name: pretty_site_name,
+        title: @json_content["title"],
+        created_at: @json_content["creation_date"],
+        score: @json_content["score"],
+        comment_count: @json_content["comment_count"],
+        answer_count: @json_content["answer_count"],
+        post_type: @post_type,
+        post_url: @json_content["link"],
+        body: @json_content["body"]
+      },
+    )
+  end
+
+  private
+
+  def valid_site?(site)
+    (site =~ /[a-z.]+/i)&.zero?
+  end
+
+  def parse_site(input)
+    return "stackoverflow" if tag_name == "stackoverflow"
+
+    site = input.match(/[a-z.]+/i)[0]
+    raise StandardError, "Invalid Stack Exchange site: {% #{tag_name} #{input} %}" unless valid_site?(site)
+
+    site
+  end
+
+  def valid_id?(input)
+    (input =~ /\A\d{1,10}\Z/i)&.zero?
+  end
+
+  def handle_response_errors(responses_array)
+    responses_array.map do |response|
+      raise StandardError, "Calling StackExchange API failed: #{response&.error_message}" if response.code != 200
+    end
+  end
+
+  def get_data(input)
+    id = input.match(/\d+/i)[0]
+    raise StandardError, "Invalid Stack Exchange ID: {% #{tag_name} #{input} %}" unless valid_id?(id)
+
+    post_response = HTTParty.get("#{API_URL}posts/#{id}?site=#{@site}&filter=#{FILTERS['post']}&key=#{ApplicationConfig['STACK_EXCHANGE_APP_KEY']}")
```

Comment:
`HTTParty.get` can potentially raise an exception or return a response that doesn't have some part of the `["items"][0]["post_type"]`. I think it's worth handling these cases to avoid showing 500 to the user.

Reply:
Here are the types of errors you will need to plan for:

- Any network level stuff (timeout, etc).
- Any non-200 HTTP status code is an API failure.
- A 200 status code with HTML content.  This happens when Stack Exchange is having an outage the CDN/load balancers point to the error page.
- A response where [`error_id`](https://api.stackexchange.com/docs/error-handling) exists.  The routes you're hitting _shouldn't_ generally produce errors, but there are things [like integers we can't parse](https://api.stackexchange.com/docs/questions-by-ids#order=desc&sort=activity&ids=9999999999999999999999999999999999999&filter=default&site=stackoverflow&run=true) and [rate limiting](https://api.stackexchange.com/docs/throttle) to worry about.  Given the rarity of these, and that they probably indicate a bug the developer has to fix, I'd log the response and show a "Couldn't load post" message to the user.
- `["items"].length == 0`.  This will by far be the most common issue.  This comes up if we can't find a non-deleted post with that ID (or on /questions if the post is an answer).  For example, [`{% stackoverflow 1 %}`](https://api.stackexchange.com/docs/posts-by-ids#order=desc&sort=activity&ids=1&filter=default&site=stackoverflow&run=true)

Reply:
I currently have some error handling in the `handle_response_error` method, and it should account for:
- network level issues
- non-200 HTTP status codes
- `["items"].length == 0`

It does **not** handle `error_id`, but I think my implementation should prevent non-integers from being sent to the API.

Reply:
Does it handle network-level exceptions? As I see, if an exception is raised while executing `HTTParty`, it won't be rescued. `handle_response_error` will run only if the code before runs without exceptions.